sanitize.go

```diff
@@ -33,11 +33,14 @@ import (
 	"bytes"
 	"io"
 	"net/url"
+	"regexp"
 	"strings"
 
 	"golang.org/x/net/html"
 )
 
+var dataAttribute = regexp.MustCompile("^data-.+$")
```

Comment:
Spec says that the name part of a data-* attribute must conform to a number of things, this regexp should enforce that... the .* bit is wrong.